File: src/main/java/hacker/solution/Prime.java
package hacker.solution;

class Prime {
    void checkPrime(int... numbers) {
        for (int num : numbers) {
            if (isPrime(num)) {
                System.out.print(num + " ");
            }
        }
        System.out.println();
    }
    boolean isPrime(int n){
        if (n == 2)
            return true;
        if (n < 2 || n % 2 == 0)
            return false;
        for (int i = 3; i <= (int) Math.sqrt(n); i += 2){
            if (n % i == 0)
                return false;
        }
        return true;
    }
}

File: src/main/kotlin/hacker/solutionk/Prime.kt
package hacker.solutionk

class Prime {
    fun checkPrime(vararg numbers: Int) {
        for (num in numbers) {
            if (isPrime(num)) {
                print("$num ")
            }
        }
        println()
    }
    fun isPrime(n: Int): Boolean {
        if (n < 2) {
            return false
        } else if (n == 2) {
            return true
        } else if (n % 2 == 0) {
            return false
        }
        val sqrt = Math.sqrt(n.toDouble()).toInt()
        var i = 3
        while (i <= sqrt) {
            if (n % i == 0) {
                return false
            }
            i += 2
        }
        return true
    }
}
